### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by polypx**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -70,29 +70,440 @@ write 3 UDTs below that EACH have:
...
+        population = population + immigrantsYear - emigrantsYear + birthsYear - deathsYear;
+    }    
+    return population;
+}
+
+float City::PoliceDepartment::getConvictionRate(float arrests, float convictions)
+{
+    std::cout << convictions / arrests << " is the conviction rate." << std::endl;
+    return convictions / arrests;
+}
```
Comment (matkatmusic) [inline main.cpp]: D.R.Y. 
you're computing this value twice. 
cache it into a local variable. 
use it in the std::cout statement and the return statement
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -70,29 +70,440 @@ write 3 UDTs below that EACH have:
...
+{
+    lightsCurrentState = !lightsCurrentState;
+    if (lightsCurrentState)
+    {
+        std::cout << "The lights are currently on."  << std::endl; 
+    } 
+    else
+    {
+        std::cout << "The lights are currently off."  << std::endl; 
+    }
```
Comment (matkatmusic) [inline main.cpp]: You can rewrite this as:
```
std::cout << "The lights are currently " << (lightsCurrentState ? "on." : "off.") << std::endl;
```
This makes use of a Ternary Expression. (check cppreference to learn about them)
Try it out. 
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -70,29 +70,440 @@ write 3 UDTs below that EACH have:
...
+int LargestFiveCities::setPopulations(int a, int b, int c, int d, int e)
+{
+    cityA.population = a;
+    cityB.population = b;
+    cityC.population = c;
+    cityD.population = d;
+    cityE.population = e;
+
+    std::cout << "Combined population is " << a + b + c + d + e << std::endl; 
+    return a + b + c + d + e;
```
Comment (matkatmusic) [inline main.cpp]: D.R.Y. 
cache this computed value locally
use it in the std::cout and return statements
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 2 above)
Comment (polypx) [reply main.cpp]: Excellent, I used this is a few places to tighten the code a bit.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**